axelor-prestashop/src/main/java/com/axelor/apps/prestashop/exports/service/ExportProductServiceImpl.java

```diff
@@ -419,32 +421,30 @@ private void exportStocks(
     int done = 0;
     logBuffer.write(String.format("%n===== STOCKS =====%n"));
 
-    @SuppressWarnings("unchecked")
-    final List<Object[]> stocks =
-        JPA.em()
-            .createQuery(
-                "SELECT product, "
-                    + "("
-                    + "SELECT COALESCE(SUM(CASE WHEN fromLocation.typeSelect = :virtualLocation THEN line.realQty ELSE -line.realQty END), 0) "
-                    + "FROM StockMoveLine line "
-                    + "JOIN line.stockMove move "
-                    + "JOIN move.fromStockLocation fromLocation "
-                    + "JOIN move.toStockLocation toLocation "
-                    + "WHERE line.product = product "
-                    + "AND move.statusSelect != :canceledStatus "
-                    + "AND (fromLocation.typeSelect != :virtualLocation OR toLocation.typeSelect != :virtualLocation) "
-                    + "AND (fromLocation.typeSelect = :virtualLocation OR toLocation.typeSelect = :virtualLocation) "
-                    + ")"
-                    + "FROM Product product "
-                    + "WHERE product.prestaShopId is not null "
-                    + "GROUP BY product")
-            .setParameter("canceledStatus", StockMoveRepository.STATUS_CANCELED)
-            .setParameter("virtualLocation", StockLocationRepository.TYPE_VIRTUAL)
-            .getResultList();
-    for (Object[] row : stocks) {
+    List<StockLocationLine> stockLocationLineList =
+        Beans.get(StockLocationLineRepository.class)
+            .all()
+            .filter(
+                "self.product.prestaShopId IS NOT NULL AND self.stockLocation.typeSelect != ?1",
+                StockLocationRepository.TYPE_VIRTUAL)
+            .fetch();
+
```

Comment:
Use location service to get real qty of the product, pass null value for location.